src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,207 @@
+                    firstPart.append(desc[1]);
+                } else {
+                    for (int i = 1; i < desc.length; i++) {
+                        firstPart.append(desc[i]);
+                        if (i < desc.length - 1) {
+                            firstPart.append(command + " ");
+                        }
+                    }
+                }
+                for (int i = 1; i < inputs.length - 1; i++) {
+                    if (i < inputs.length - 1) {
+                        firstPart.append(" " + format + " ");
+                    }
+                    firstPart.append(inputs[i]);
+                }
+                params[0] = firstPart.toString();
+                params[1] = inputs[inputs.length - 1];
+            }
+            return params;
+        } catch (PatternSyntaxException e) {
+            throw new WrongInputException("D: D: D: Please specify the timing using " + format + " D: D: D:");
+        }
+    }
+
+    /**
+     * Returns the task description contained in the user input.
+     *
+     * @param s The user input.
+     * @param command The command contained in the user input.
+     * @return The task description.
+     * @throws IncompleteInputException If the input is incomplete.
+     */
+    public String parseStringDescription(String s, String command) throws IncompleteInputException {
+        String[] inputs = s.split(command + " ");
+        if (inputs.length == 1) {
+            throw new IncompleteInputException("D: D: D: The description of a " + command
+                    + " cannot be empty D: D: D:");
+        }
+        if (inputs.length == 2) {
+            return inputs[1];
+        }
+        StringBuilder output = new StringBuilder();
+        for (int i = 1; i < inputs.length; i++) {
+            output.append(inputs[i]);
+            if (i < inputs.length - 1) {
+                output.append(command + " ");
+            }
+        }
+        return output.toString();
+    }
+
+    /**
+     * Returns a Task object from parsing a saved line in a storage file.
+     *
+     * @param entry A line from a storage file containing the description of a Task.
+     * @return A Task object formed from parsing a line in a storage file.
+     * @throws DateTimeParseException If any time in the file is not in the correct format.
+     */
+    public static Task parseFileFormat(String entry) throws DateTimeParseException {
+        String[] entrySplit = entry.split(" \\| ");
+        if (entrySplit[0].equals("T")) {
+            if (entrySplit[1].equals("1")) {
+                return new Todo(entrySplit[2], true);
+            } else {
+                return new Todo(entrySplit[2], false);
+            }
+        } else if (entrySplit[0].equals("D")) {
+            if (entrySplit[1].equals("1")) {
+                return new Deadline(entrySplit[2],
+                        LocalDate.parse(entrySplit[3], DateTimeFormatter.ofPattern("MMM dd yyyy")),
```

Comment:
I like the 8 spaces for the indentation for the wrapped line. Great adherence to the coding standard.